### PR TITLE
Add missing BTI bits for fcontext use

### PR DESCRIPTION
**Pull request by lgv5**

### CODE REVIEW

File: src/fcontext/fcontext.c [hunk 1]
```diff
@@ -4,6 +4,21 @@
 #include <stdlib.h>
 #include <assert.h>
 
+// Lifted from https://github.com/llvm/llvm-project/blob/release/16.x/clang/lib/Headers/cet.h
+# ifdef __LP64__
+#  if __CET__ & 0x1
```
Comment (Akaricchi) [inline src/fcontext/fcontext.c:12]: Need to test for `#ifdef __CET__` before trying to use it in an expression.

File: src/fcontext/fcontext.c [hunk 2]
```diff
@@ -4,6 +4,21 @@
...
+#    define _CET_ENDBR __asm("endbr64")
+#  else
+#    define _CET_ENDBR
```
Comment (Akaricchi) [inline src/fcontext/fcontext.c]: The empty stub definition should be `#define _CET_ENDBR ((void)0)`. This forces the compiler to require a semicolon after the macro and silences potential warnings about "extraneous" semicolons.

**Review by Akaricchi — CHANGES_REQUESTED** [review]
LGTM except for some nitpicks

File: src/fcontext/fcontext.c (hunk 1 above)
Comment (lgv5) [reply src/fcontext/fcontext.c:12]: yes, my bad.

File: src/fcontext/fcontext.c (hunk 2 above)
Comment (lgv5) [reply src/fcontext/fcontext.c]: Are you sure about this one? I don't get any complain with or without a definition after `_CET_ENDBR`.

Comment (Akaricchi) [reply src/fcontext/fcontext.c]: It depends on your compiler and warning flags. I find it to be good practice even if there is no warning, just to make the compiler consistently require the semicolon.

**Comment by lgv5** [comment]
Sorry for the last 2 force-pushes, I screwed up my author information.

**Review by Akaricchi — APPROVED** [review]

**Comment by Akaricchi** [comment]
Thanks!

**Comment by Akaricchi** [comment]
Also merged into master now.

**Pull request merged**